File: Week2/Exercise_2.2.py
"""
Author:     Alan Danque
Date:       20200913
Exercise:   2.2
"""


import matplotlib.pyplot as plt; plt.rcdefaults()
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import warnings
import matplotlib.cbook
warnings.filterwarnings("ignore",category=matplotlib.cbook.mplDeprecation)

#Step 1:  Load data into the dataframe
ppgdf = pd.read_csv(".\\datasets\\ppg2008.csv")
ppgdf_sorted = ppgdf.sort_values('G',ascending=True)
ppgdf_sorted.insert(0, 'ID', range(0, 0 + len(ppgdf_sorted)))
print(ppgdf_sorted)

#Line Chart
ppgdf_sorted.plot(x ='ID', y='G', kind = 'line', color="blue")
plt.legend(["Games Played"]);
plt.xlabel("Player ID", size=15)
plt.ylabel("Games Played", size=15)
plt.title("Number of Games Played by Player ID", size=18)
plt.savefig(".\\plots\\Line Chart Number of Games Played by Player ID Python.png")
plt.show()
plt.close()

#Step Charts
plt.figure()
list_id = ppgdf_sorted['ID'].to_list()
g_id = ppgdf_sorted['G'].to_list()
plt.plot(list_id, g_id, drawstyle='steps', linestyle='-', label='steps-pre aka steps', alpha=1, color="grey")
plt.legend(["Games Played"]);
plt.xlabel("Player ID", size=15)
plt.ylabel("Games Played", size=15)
plt.title("Number of Games Played by Player ID", size=18)
plt.savefig(".\\plots\\Step Chart Number of Games Played by Player ID Python.png")
plt.show()
plt.close()

File: Week2/Exercise_2.2.r
#Author: Alan Danque
#Date:   20200913
#Class:  DSC 640
#Exercise 2.2

#install.packages("rpivotTable")
#install.packages("readxl")
#install.packages("tidyverse")
#install.packages("reshape2")
#library(reshape2)
#library(rpivotTable)

# create list of packages we need 
packages <- c("ggplot2", "dplyr", "gapminder", "readxl", "tidyverse", "reshape2")
print(packages)
# Install packages 
lapply(packages, install.packages, character.only = TRUE)

# Load packages
lapply(packages, library, character.only = TRUE)
# In case you are unfamiliar with lapply() - it has been used to apply the install.packages() and library() functions over a list of package names. More information here: https://www.r-bloggers.com/using-apply-sapply-lapply-in-r/

# Read csv input
pggdata <- read.csv("./datasets/ppg2008.csv")

# Sort the dataframe on Games Played
pggdata_sorted <- pggdata[with (pggdata, order(G)),]

# Add unique ID
pggdata_sorted$ID <- 1:nrow(pggdata_sorted) 

print(is.data.frame(pggdata_sorted))
print(ncol(pggdata_sorted))
print(nrow(pggdata_sorted))

IDvector <- as.vector(pggdata_sorted$ID)
Gvector <- as.vector(pggdata_sorted$G)

print(IDvector)
print(Gvector)

print(typeof(IDvector))
print(typeof(Gvector))
dev.off()

# Line Chart
plot(IDvector,Gvector, type="l", col="blue", lwd=5, xlab="Player ID", ylab="Games Played", main="Number of Games Played by Player ID")
dev.copy(png,"./plots/Line Chart Number of Games Played by Player ID R.png")
dev.off()

# Step Chart
plot(IDvector,Gvector, type="s", col="grey", lwd=5, xlab="Player ID", ylab="Games Played", main="Number of Games Played by Player ID")
dev.copy(png,"./plots/Step Chart Number of Games Played by Player ID R.png")
dev.off()



